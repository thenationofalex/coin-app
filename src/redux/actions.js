export function syncApi() {
  return {
    type: 'SYNC_API'
  }
}
