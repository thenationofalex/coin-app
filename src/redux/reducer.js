const initialState = {
  coinData: 'loading',
  error: false,
  loading: true
}

const reducer = (state = initialState, action) => {
  switch (action.type) {
  case 'SYNC_API':
    return state
  case 'STORE_REMOTE_DATA':
    return {
      error: false,
      loading: false,
      coinData: action.data
    }
  case 'STORE_REMOTE_ERROR':
    return {
      loading: false,
      error: true,
      coinData: 'Sorry the connection to the API failed. Please wait a few moments and refresh the page.'
    }
  default:
    return state
  }
}

export default reducer
