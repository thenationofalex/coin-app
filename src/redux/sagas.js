import { takeEvery, call, put } from 'redux-saga/effects'
import { apiData } from '../api'

function * fetchCoinData() {
  try {
    const callApi = yield call(apiData)
    yield put({ type: 'STORE_REMOTE_DATA', data: callApi })
  } catch (e) {
    yield put({ type: 'STORE_REMOTE_ERROR' })
  }
  
}

function * syncSagas() {
  yield takeEvery('SYNC_API', fetchCoinData)
}

export default syncSagas
