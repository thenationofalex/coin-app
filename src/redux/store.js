import { createStore, combineReducers, applyMiddleware } from 'redux'
import createSagaMiddleware from 'redux-saga'
import createHistory from 'history/createBrowserHistory'
import { routerReducer, routerMiddleware } from 'react-router-redux'
import reducer from './reducer'
import syncSagas from './sagas'

export const history = createHistory()
const sagaMiddleware = createSagaMiddleware()
const routeMiddle = routerMiddleware(history)

export const store = createStore(
  combineReducers({
    reducer,
    router: routerReducer
  }),
  applyMiddleware(sagaMiddleware, routeMiddle)
)

sagaMiddleware.run(syncSagas)
