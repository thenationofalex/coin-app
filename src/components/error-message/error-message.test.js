import React, {Component} from 'react'
import {shallow} from 'enzyme'
import toJson from 'enzyme-to-json'

import { store } from '../../redux/store'
import {ErrorMessage} from './index'
 
test('ErrorMessage renders', () => {
  const props = {
    store: store,
    theme: {
      colors: {
        blue: 'blue'
      }
    }
  }
  const wrapper = shallow(
    <ErrorMessage {...props}>
      ErrorMessage
    </ErrorMessage>
  );
 
  expect(toJson(wrapper)).toMatchSnapshot()
});
