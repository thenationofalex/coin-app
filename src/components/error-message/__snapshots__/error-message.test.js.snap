// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorMessage renders 1`] = `
<div
  className="css-vpaqbd"
>
  ErrorMessage
</div>
`;
